static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -0,0 +1,90 @@
+                    headerHeight = $header.height();
+
+                var topAdRenderedPromise = new Promise(function (resolve) {
+                    mediator.on('modules:commercial:dfp:rendered', function (event) {
+                        if (event.slot.getSlotElementId() === adId) {
+                            resolve();
+                        }
+                    });
+                });
+
+                var oldAdHeightPromise = Promise.resolve(oldAdHeight);
+                var newAdHeightPromise = topAdRenderedPromise.then(function () {
+                    return fastdom.read(function () {
+                        var adSlotPadding = parseInt($('#' + adId, $adBanner).css('padding-bottom')) * 2;
+                        // We must read the iframe attribute height to avoid reading the clientHeight mid-transition
+                        return Number($('iframe', $adBanner).attr('height')) + adSlotPadding;
+                    });
+                });
+
+                var getLatestAdHeight = function () {
+                    return Promise.race([newAdHeightPromise, oldAdHeightPromise]);
+                };
+
+                var render = function () {
+                    getLatestAdHeight().then(function (adHeight) {
+                        fastdom.write(function () {
+                            if (window.scrollY === 0) {
+                                // Reset
+                                $header.css('margin-top', '');
+                                $adBanner.css({
+                                    'position': '',
+                                    'top': ''
+                                });
+                            } else {
+                                if (window.scrollY > (headerHeight)) {
+                                    $adBanner.css({
+                                        'position': 'absolute',
+                                        'top': headerHeight + 'px'
+                                    });
+                                } else {
+                                    $adBanner.css({
+                                        'position': 'fixed',
+                                        'top': 0
+                                    });
+                                }
+                                $header.css('margin-top', adHeight + 'px');
+                            }
+                        });
+                    });
+                };
+
+                //
+                // Side effects
+                //
+
+                mediator.on('window:throttledScroll', render);
+                render();
+                topAdRenderedPromise.then(render);
+
+                newAdHeightPromise.then(function (adHeight) {
+                    var diff = adHeight - oldAdHeight;
+
+                    if (window.scrollY !== 0) {
+                        window.scrollTo(window.scrollX, window.scrollY + diff);
+                    }
+                });
```

Comment:
The above isn't actually tied to the sticky ad, it's just part of the top ad itself. Should we move it?